src/core/model.js

```diff
@@ -326,54 +325,41 @@ export default class Model extends Object3D {
     return this;
   }
 
-  // PROFILING - TODO - rebuild using Query class
   _timerQueryStart() {
-    if (this.timerQueryEnabled === true && this.lastQueryReturned === true && this.ext) {
-      this.program.gl.getParameter(this.ext.GPU_DISJOINT_EXT);
-      this.timeElapsedQuery = this.ext.createQueryEXT();
-      this.ext.beginQueryEXT(this.ext.TIME_ELAPSED_EXT, this.timeElapsedQuery);
+    if (this.timerQueryEnabled === true) {
+      if (!this.timeElapsedQuery) {
+        this.timeElapsedQuery = new Query(this.gl);
+      }
+      if (this.lastQueryReturned) {
```

Comment:
Is there a method or member on the query object we could use instead of keeping this extra state here: `lastQueryReturned`?

The point of the `Query` class is to simplify use in apps, but we it seems that we still need to do a lot of book keeping here.